src/npctalk.cpp

```diff
@@ -2302,6 +2302,7 @@ static void receive_item( itype_id &item_name, int count, const std::string &con
     }
     if( container_name.empty() ) {
         if( new_item.count_by_charges() ) {
+            new_item.charges = 1;    // Reset item default charges, prepare to receive specific defined charges
             new_item.mod_charges( count - 1 );
```

Comment:
```suggestion
            new_item.charges = count;
```

Might as well simplify it.
I also see the same thing a bit further down for items that come in a container.

Reply:
Yes! Thank you.